lib/handler/pool/state.ex

```diff
@@ -14,15 +14,23 @@ defmodule Handler.Pool.State do
           running_workers: non_neg_integer(),
           bytes_committed: non_neg_integer(),
           workers: %{
-            reference() => {
-              bytes_committed :: non_neg_integer(),
-              from_pid :: pid(),
-              task_pid :: pid(),
-              task_name :: String.t()
-            }
+            reference() => worker()
           },
           pool: %Handler.Pool{}
         }
+  @type worker :: local_worker() | delegated_worker()
+  @type local_worker :: %{
+          bytes_committed: non_neg_integer(),
+          from_pid: pid(),
```

Comment:
Bit of a nit-pick, but it might be nice to reorder these fields so that it's more immediately apparent where the difference is. E.g.

```elixir
@type worker :: local_worker() | delegated_worker()
@type local_worker :: %{
        bytes_committed: non_neg_integer(),
        from_pid: pid(),
        task_name: String.t() | nil,
        task_pid: pid()
      }
@type delegated_worker :: %{
        bytes_committed: non_neg_integer(),
        from_pid: pid(),
        task_name: String.t() | nil,
        delegated_to: Pool.pool()
      }
```

Reply:
👍 I like that idea